File: Listas, Colas y Pilas/Pilas/PilaArreglo.py
class PilaArreglo:
    def __init__(self, tamano_max):
        self.stack = [None] * tamano_max
        self.max_size = tamano_max
        self.top = -1

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.max_size - 1

    def push(self, data):
        if self.isFull():
            print("Error: Stack Overflow")
            return
        self.top += 1
        self.stack[self.top] = data

    def pop(self):
        if self.isEmpty():
            print("Error: Stack Underflow")
            return -1
        data = self.stack[self.top]
        self.top -= 1
        return data

    def peek(self):
        if self.isEmpty():
            print("Pila vacía")
            return -1
        return self.stack[self.top]

if __name__ == "__main__":
    pila = PilaArreglo(100)
    pila.push(10)
    pila.push(20)
    pila.push(30)
    
    print(f"Elemento superior: {pila.peek()}")
    print(f"Extrae elemento: {pila.pop()}")
    print(f"Nuevo elemento superior: {pila.peek()}")

File: Listas, Colas y Pilas/Pilas/PilaArreglo.cs
using System;

class PilaArreglo {
    private int[] stack;
    private int maxSize;
    private int top;

    public PilaArreglo(int tamanoMax) {
        maxSize = tamanoMax;
        stack = new int[maxSize];
        top = -1;
    }

    public bool IsEmpty() {
        return top == -1;
    }

    public bool IsFull() {
        return top == maxSize - 1;
    }

    public void Push(int data) {
        if (IsFull()) {
            Console.WriteLine("Error: Stack Overflow");
            return;
        }
        stack[++top] = data;
    }

    public int Pop() {
        if (IsEmpty()) {
            Console.WriteLine("Error: Stack Underflow");
            return -1;
        }
        return stack[top--];
    }

    public int Peek() {
        if (IsEmpty()) {
            Console.WriteLine("Pila vacía");
            return -1;
        }
        return stack[top];
    }
}

class Program {
    static void Main() {
        PilaArreglo pila = new PilaArreglo(100);
        pila.Push(10);
        pila.Push(20);
        pila.Push(30);

        Console.WriteLine("Elemento superior: " + pila.Peek());
        Console.WriteLine("Extrae elemento: " + pila.Pop());
        Console.WriteLine("Nuevo elemento superior: " + pila.Peek());
    }
}
